src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/design_patterns/StrategyDesignPattern.java

```diff
@@ -0,0 +1,30 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.design_patterns;
+
+public class StrategyDesignPattern extends DesignPattern {
+    StrategyDesignPattern(){}
+
+    @Override
+    public String getName() {
+        return "Strategy";
+    }
+
+    @Override
+    public Type getType() {
+        return Type.BEHAVIORAL;
+    }
+
+    @Override
+    public String getDescription() {
+        return "some strategy desc";
+    }
+
+    @Override
+    public String getExample() {
+        return "some strategy example";
+    }
+
+    @Override
+    public RefactorProvider createRefactorProvider() {
+        return badCode -> badCode + " now improved with Strategy DP";
```

Comment:
Why not simply:
`public String refactor()`
Why separate design pattern from refactoring strategy. What's the gain with this longer line in library usage:
`.createRefactorProvider().refactor("if false comment='why?'");`